### PR TITLE
New staking component

### PR DESCRIPTION
**Pull request by bobo-k2**
**Pull Request Summary**

This PR is to create a new staking component (wizard style). The component can be used from multiple pages. Currently I set the component to be used from dApp staking discover page and from the staking page.

The PR goes to [feat/dapp-staking-v3-period2](https://github.com/AstarNetwork/astar-apps/tree/feat/dapp-staking-v3-period2) feature branch. Plan is to merge and release the feature branch sometimes before new voting period on Astar starts

image

image

**Check list**

- [ ] contains breaking changes
- [x] adds new feature
- [x] modifies existing feature (bug fix or improvements)
- [ ] relies on other tasks
- [ ] documentation changes
- [ ] tested on mobile devices

### CODE REVIEW

File: src/staking-v3/components/vote/VotingWizard.vue [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+  },
+  setup(props) {
+    const { t } = useI18n();
+    const { selectedComponentIndex, handleSelectComponent } = useSelectableComponent();
+    const { registeredDapps, getDapp } = useDapps();
+    const { nativeTokenImg } = useNetworkInfo();
+    const selectedDapps = ref<DappVote[]>([]);
+    const stakesEntered = ref<boolean>(false);
+    const isConfirmed = ref<boolean>(false);
+    const wizard = ref();
```
Comment (impelcrypto) [inline src/staking-v3/components/vote/VotingWizard.vue:113]: How `wizard` state be updated? I can't find `wizard.value =` in this file.

File: src/staking-v3/components/vote/choose-dapps/ChooseCategory.vue [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  props: {
+    onCategorySelected: {
+      type: Function as PropType<(category: string) => void>,
+      required: true,
+    },
+  },
+  setup(props) {
+    const { isComponentSelected, handleSelectComponent } = useSelectableComponent();
+
+    const handleCategorySelected = (index: number) => {
```
Comment (impelcrypto) [inline src/staking-v3/components/vote/choose-dapps/ChooseCategory.vue:39]:
```suggestion
    const handleCategorySelected = (index: number): void => {
```

File: src/staking-v3/components/vote/enter-amount/ChooseAmountsPanel.vue [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+        props.onRemoveDapp(dapp);
+      }
+    };
+
+    return { handleAmountChanged, handleRemoveDapp };
+  },
+});
+</script>
+
+<style lang="scss" scoped>
```
Comment (impelcrypto) [inline src/staking-v3/components/vote/enter-amount/ChooseAmountsPanel.vue:139]: I think the number of lines in this file is a bit too many; perhaps it's better to separate the styling code from this file (and some other files too).

**Review by impelcrypto — APPROVED** [review]

File: src/staking-v3/components/vote/enter-amount/ChooseAmountsPanel.vue (hunk 3 above)
Comment (bobo-k2) [reply src/staking-v3/components/vote/enter-amount/ChooseAmountsPanel.vue:130]: Thanks for comments. Fixed in https://github.com/AstarNetwork/astar-apps/pull/1291/commits/200eee81e648cad59d98ff69aae9658a649833c6

**Review by gtg7784 — APPROVED** [review]

**Pull request merged**